### ISSUE TITLE
Add random.binomial and random.multinomial

### ISSUE THREAD

**Issue by carlosgmartin** [jax-ml/jax#13327]
Add JAX counterparts of [numpy.random.binomial](https://numpy.org/doc/stable/reference/random/generated/numpy.random.binomial.html) and [numpy.random.multinomial](https://numpy.org/doc/stable/reference/random/generated/numpy.random.multinomial.html) to [jax.random](https://jax.readthedocs.io/en/latest/jax.random.html) package. See https://github.com/google/jax/issues/480#issuecomment-1272587620 for context. A current workaround is using the [JAX substrate](https://www.tensorflow.org/probability/api_docs/python/tfp/substrates/jax) of [TensorFlow Probability](https://www.tensorflow.org/probability):
```python3
from jax import random, numpy as jnp
from tensorflow_probability.substrates import jax as tfp

def binomial(key, n, p, shape=()):
    return tfp.distributions.Binomial(n, probs=p).sample(
        seed=key,
        sample_shape=shape,
    )

def multinomial(key, n, p, shape=()):
    return tfp.distributions.Multinomial(n, probs=p).sample(
        seed=key,
        sample_shape=shape,
    )

key = random.PRNGKey(0)

key, subkey = random.split(key)
print(binomial(subkey, 9, .8, [2, 5]))

key, subkey = random.split(key)
print(multinomial(subkey, 9, jnp.array([.7, .2, .1]), [4]))
```
Output:
```
[[7. 8. 8. 7. 5.]
 [5. 8. 9. 8. 5.]]
[[7. 1. 1.]
 [4. 3. 2.]
 [5. 3. 1.]
 [4. 4. 1.]]
```

**Comment by zhangqiaorjc** [comment]
@sharadmv, is it possible to port tfp's implementation to jax.random?

**Comment by hylkedonker** [comment]
Any plans to add `random.multinomial` now that `random.binomial` has been merged? 
The distribution could be implemented either as sequence of negative binomials or by repeated categorical draws. 
See also the Wikipedia entry on [multinomial variate generation](https://en.wikipedia.org/wiki/Multinomial_distribution#Random_variate_generation)).

**Comment by jakevdp** [comment]
Thanks for reaching out – I don't know of anyone working on this currently.

**Comment by andportnoy** [comment]
A workaround if you want pure JAX is to take the log of your probabilities vector (that sums to 1):
```python
jax.random.categorical(key, jnp.log(p))
```
(Based on the "contract" of `jax.random.categorical`:
> logits: Unnormalized log probabilities of the categorical distribution(s) to sample from,
      so that `softmax(logits, axis)` gives the corresponding probabilities.

)

That can't be the most efficient way to sample though...

**Comment by brianwa84** [comment]
Fyi: You would need to sum many categorical samples to get a multinomial
sample.

**Comment by ShawnGeller** [comment]
I would use a multinomial feature if added! The categorical workaround is a nonstarter if I want a lot of samples.

**Comment by yoavram** [comment]
👍

**Comment by cthorrez** [comment]
I would benefit from multinomial as well. I'm currently using the numpy one for efficient bootstrap sampling and not having it here makes it harder to make the switch to jax.

**Comment by carlosgmartin** [comment]
I've created a PR for this: https://github.com/jax-ml/jax/pull/25688.

**Comment by jakevdp** [comment]
Re-opening, as #25688 had to be rolled back due to TPU test failures.
